src/docs/sphinx/dev_guide/gpu_porting.rst

```diff
@@ -0,0 +1,302 @@
+.. ## Copyright (c) 2017-2022, Lawrence Livermore National Security,con LLC and
+.. ## other Axom Project Developers. See the top-level LICENSE file for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+
+.. _gpu-porting-label:
+
+*******************************
+GPU Porting in Axom
+*******************************
+
+Axom uses RAJA and Umpire as the main workhorses for GPU porting,
+with a set of convenience macros and `axom`-namespaced wrappers encapsulating
+commonly used RAJA and Umpire functions, and preset execution spaces for
+host/device execution.
+
+For the user's guide on using GPU utilities, see also
+:ref:`Core Acceleration<core-acceleration>`
+
+
+.. _gpu-macros-label:
+
+===============
+Macros 
+===============
+
+Axom's macros can be found in file
+`axom/core/Macros.hpp <https://github.com/LLNL/axom/blob/develop/src/axom/core/Macros.hpp>`_
+
+Most of the GPU-related macros are used to guard device code for compilation
+time or for ``__host__ __device__`` decoration of functions/lambdas.
+
+For guarding device code:
+
+.. literalinclude:: ../../../axom/core/Macros.hpp
+   :start-after:  _guarding_macros_start
+   :end-before:  _guarding_macros_end
+   :language: C++
+
+For ``__host__`` or `` __device__`` decoration, or both:
```

Comment:
```suggestion
The following code shows the macros used in Axom to apply these annotations.
```
Also, it may be a good idea to include a code example showing how these macros are used.